contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -79,141 +193,75 @@ function syncChanges() {
-            const errorMap = {};
-            let errorSetPromise = Promise.resolve();
-            if (errors.length) {
-              for (let error of errors) {
-                errorMap[error.rev] = error;
-              }
-              // Set the return error data onto the changes - this will update the change
-              // both with any errors and the results of any merging that happened prior
-              // to the sync operation being called
-              errorSetPromise = db[CHANGES_TABLE].where('rev')
-                .anyOf(Object.keys(errorMap).map(Number))
-                .modify(obj => {
-                  return Object.assign(obj, errorMap[obj.rev]);
-                });
-            }
-            let changesToDelete;
-            if (!allErrors) {
-              changesToDelete = db[CHANGES_TABLE].where('rev').belowOrEqual(changesMaxRevision);
-              if (errors.length) {
-                // Filter changes by whether the revision for this change is in the errorMap
-                // merged changes will have been returned from the server whole, so deleting
-                // older changes will not be a problem here, as we maintain the merged
-                // representation
-                changesToDelete = changesToDelete.filter(change => !errorMap[change.rev]);
-              }
-            }
-            const deleteChangesPromise =
-              lastChange && !allErrors ? changesToDelete.delete() : Promise.resolve();
-            const returnedChangesPromise = returnedChanges.length
-              ? applyChanges(returnedChanges)
-              : Promise.resolve();
-            // Our synchronization was successful,
-            // can delete all the changes for this table
-            return Promise.all([
-              deleteChangesPromise,
-              returnedChangesPromise,
-              errorSetPromise,
-            ]).catch(() => {
-              console.error('There was an error deleting changes'); // eslint-disable-line no-console
-            });
-          })
-          .catch(err => {
-            // There was an error during syncing, log, but carry on
-            console.warn('There was an error during syncing with the backend for', err); // eslint-disable-line no-console
-          });
-      });
-    })
-    .then(() => {
-      syncActive = false;
-    })
-    .catch(() => {
-      syncActive = false;
-    });
+  const [lastChange, user] = await Promise.all([
+    db[CHANGES_TABLE].orderBy('rev').last(),
+    Session.getSession(),
+  ]);
+  if (!user) {
+    // If not logged in, nothing to do.
+    return;
+  }
+
+  const now = Date.now();
+  const channel_ids = Object.entries(user[ACTIVE_CHANNELS] || {})
+    .filter(([id, time]) => id && time > now - CHANNEL_SYNC_KEEP_ALIVE_INTERVAL)
+    .map(([id]) => id);
+  const channel_revs = {};
+  for (let channelId of channel_ids) {
+    channel_revs[channelId] = get(user, [MAX_REV_KEY, channelId], 0);
```

Comment:
Should this be 
```js
get(user, `${MAX_REV_KEY}.${channelId}`, 0)
```
???

Reply:
This is the same, the lodash `get` function allows you to specify the dot path either as a dot path or an array.

Reply:
But doesn't the dot path lookup follow nesting so the structure would have to be `{[MAX_REV_KEY]: { [channelId]: ... } }`? The key I see you setting is flat, not nested

Reply:
I guess my suggestion probably wouldn't work either with lodash `get`

Reply:
Ah, right - possible that I did not set this properly, will revist!